### DISCUSSION TITLE
How to get Event of 'GUI start/stop record'

### DISCUSSION THREAD

**Discussion by chenxinfeng4** [aatikturk/obsws-python#61] [Q&A] [answered]
It's amazing of obs websocket. Your code is simple and useful. Further, I want to use it to synchronize obs with other DAQ system.

I see this code work for me. However, how to register that like 'Event: RecordStatusChanged'.
```
cl.start_record()
cl.stop_record()
```

**Comment by onyx-and-iris** [comment, marked as answer]
Hi! Thanks for your interest in the code.

The general approach to using events is as follows:

- Fetch the precise name of the event from the docs. In this case [RecordStateChanged](https://github.com/obsproject/obs-websocket/blob/master/docs/generated/protocol.md#recordstatechanged)
- define a function with matching name, snake cased and starting with `on_`. In this case `on_record_state_changed`
- register it with the event client using {obs.EventClient}.callback.register()

Something [like this](https://gist.github.com/onyx-and-iris/4b824ee93ac9dbbd701e069ee133b306) should work for your use case.

**Comment by chenxinfeng4** [comment]
Great, it really works. Nice of you!
